src/compute/src/rpc/service/stream_service.rs

```diff
@@ -36,6 +38,14 @@ impl StreamServiceImpl {
     pub fn new(mgr: LocalStreamManager, env: StreamEnvironment) -> Self {
         StreamServiceImpl { mgr, env }
     }
+
+    async fn try_get_root_actor_failure(&self) -> Option<StreamError> {
+        tokio::time::sleep(Duration::from_secs(3)).await;
```

Comment:
If a previous epoch has failed and the root failure was located, do we still need to delay by 3 seconds for each subsequent epoch? Can we improve this to let the following epochs fail immediately, by moving the `sleep` into the event loop?

Reply:
Good idea to cache the root cause. 
But blocking event loop by `sleep` will also prevent new actor failure from being added to `LocalBarrierWorker::failure_actors`, which make `sleep` useless.
Thinking about other way 🤔 .

Reply:
Just realize that there is no guarantee that, the `LocalBarrierWorker` instance being queried is the one active when actor fails 🥵 . For example:
1. An actor fails.
2.`try_get_root_actor_failure` starts to sleep.
3. Meanwhile a recovery resets the `LocalBarrierWorker`, along with failure cause.
4. An actor fails again.
5. After 3s, `try_get_root_actor_failure` queries `LocalBarrierWorker` but get wrong failure cause from step 4, not step 1.

To fix it, each `LocalBarrierWorker` instance may need an unique identifier. But I doubt it's worth doing at the moment.
@wenym1

Reply:
> Good idea to cache the root cause. But blocking event loop by `sleep` will also prevent new actor failure from being added to `LocalBarrierWorker::failure_actors`, which make `sleep` useless. Thinking about other way 🤔 .

Another solution to this can be, when there is any actor failure, in the worker loop, we keep polling the channel to receive actor failure until a 3 seconds timeout is reached. A concern is that the current barrier manager only has a single channel, and the actor failure will be mixed with other channel event, which may make this logic complicated.

Reply:
We can change to report actor error in a separated channel, if it's not complicated. Let me have a try.

Reply:
Fortunately, the sleeping for different epochs is concurrent. We may leave it as it is if we find it's hard to improve that. 🤣

Reply:
With https://github.com/risingwavelabs/risingwave/pull/15157 we can keep fetching actor error within a timeout in worker loop when getting any actor error.

Reply:
> With https://github.com/risingwavelabs/risingwave/pull/15157 we can keep fetching actor error within a timeout in worker loop when getting any actor error. This PR change to report actor error in a separate channel.

Implemented. PTAL